Zend/zend_constants.c

```diff
@@ -583,3 +567,14 @@ ZEND_API zend_result zend_register_constant(zend_constant *c)
 	}
 	return ret;
 }
+
+ZEND_API zend_result zend_register_internal_constant(const char *name, size_t name_len, zend_constant *c) {
+	zend_string *name_str = zend_string_init_interned(name, name_len, ZEND_CONSTANT_FLAGS(c) & CONST_PERSISTENT);
+
+	zend_result result = zend_register_constant(name_str, c);
+	if (result == FAILURE) {
+		zend_string_release(name_str);
```

Comment:
`zend_string_release()` doesn't do anything on interned strings.

Reply:
Actually, ignore that comment. `zend_string_init_interned()` returns non-interned strings during request with opcache enabled. We actually always need to release that string, as `zend_register_constant()` will either copy it or add it to the hash map which increases the refcount.

Reply:
> zend_string_release() doesn't do anything on interned strings.

Yeah, that was my understanding as well (https://github.com/php/php-src/pull/10954#discussion_r1154060430), but I only realised it after I pushed it (I was probably too tired yesterday).

> Actually, ignore that comment. zend_string_init_interned() returns non-interned strings during request with opcache enabled. We actually always need to release that string, as zend_register_constant() will either copy it or add it to the hash map which increases the refcount.

Ah, thanks!! I didn't check what OPCache's interned string handler does... It's very useful to know, becase I was intentionally not dealing with releasing interned strings because I had the impression that they are released any way.